File: src/프로그래머스_Level1/소수 만들기.js
function solution(nums) {
    const SIZE = 3000;
    let sosu = Array(SIZE).fill(true);
    let answer = 0;
    sosu[0] = sosu[1] = false;

    for (let i = 2; i * i < SIZE; i++) {
        if (!sosu[i]) continue;

        for (let j = i * i; j < SIZE; j += i) {
            sosu[j] = false;
        }
    }

    for (let i = 0; i < nums.length - 2; i++) {
        for (let j = i + 1; j < nums.length - 1; j++) {
            for (let k = j + 1; k < nums.length; k++) {
                let sum = nums[i] + nums[j] + nums[k];

                if (sosu[sum]) answer++;
            }
        }
    }

    return answer;
}

solution([1, 2, 3, 4]);
solution([1, 2, 7, 6, 4]);

File: src/프로그래머스_Level1/소수 만들기.py
from itertools import combinations

def prime_list(n):
    sieve = [True] * n

    m = int(n ** 0.5)
    for i in range(2, m + 1):
        if sieve[i] == True:
            for j in range(i+i, n, i):
                sieve[j] = False

    # return [i for i in range(2, n) if sieve[i] == True]
    return sieve

def solution(nums):
    prime = prime_list(3000)
    comb = list(combinations(nums, 3))
    answer = 0

    for arr in comb:
        # if sum(arr) in prime:
        if prime[sum(arr)]:
            answer += 1

    return answer

print(solution([1,2,3,4]))
print(solution([1,2,7,6,4]))
